consensus/babe/babe_test.go

```diff
@@ -211,6 +218,128 @@ func TestConfigurationFromRuntime(t *testing.T) {
 	}
 }
 
+func TestMedian_OddLength(t *testing.T) {
+	us := []uint64{3, 2, 1, 4, 5}
+	res, err := median(us)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var expected uint64 = 3
+
+	if res != expected {
+		t.Errorf("Fail: got %v expected %v\n", res, expected)
+	}
+
+}
+
+func TestMedian_EvenLength(t *testing.T) {
+	us := []uint64{1, 4, 2, 4, 5, 6}
+	res, err := median(us)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var expected uint64 = 4
+
+	if res != expected {
+		t.Errorf("Fail: got %v expected %v\n", res, expected)
+	}
+
+}
+
+func TestSlotOffset_Failing(t *testing.T) {
+	var st uint64 = 1000001
+	var se uint64 = 1000000
+
+	_, err := slotOffset(st, se)
+	if err == nil {
+		t.Fatal("Fail: did not err for c>1")
+	}
+
+}
+
+func TestSlotOffset(t *testing.T) {
+	var st uint64 = 1000000
+	var se uint64 = 1000001
+
+	res, err := slotOffset(st, se)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var expected uint64 = 1
+
+	if res != expected {
+		t.Errorf("Fail: got %v expected %v\n", res, expected)
+	}
+
+}
+
+func createFlatBlockTree_WithWaitTime(t *testing.T, depth int) *blocktree.BlockTree {
+
+	genesisBlock := types.Block{
+		Header: types.BlockHeader{
+			ParentHash: zeroHash,
+			Number:     big.NewInt(0),
+			Hash:       common.Hash{0x00},
+		},
+		Body: types.BlockBody{},
+	}
+
+	d := &db.BlockDB{
+		Db: db.NewMemDatabase(),
+	}
+
+	bt := blocktree.NewBlockTreeFromGenesis(genesisBlock, d)
+
+	// hard coded because we don't want to expose the hash of
+	previousHash := genesisBlock.Header.Hash
+
+	for i := 1; i <= depth; i++ {
+		time.Sleep(300 * time.Millisecond)
```

Comment:
Can you avoid this delay? Would be nice to keep duration of the test smaller

Reply:
we can avoid the delay and just create more blocks in the blocktree, the reason for the delay is because the slot duration in the config is 1000 milliseconds and every block will be in the same slot - 1. I can have slotTime take in slot duration as a parameter instead of pulling directly from the config of the babesession it is called on as an alternative?